### ISSUE TITLE
페이징이 안 되는데 왜 안 되는지 모르겠습니다.

### ISSUE THREAD

**Issue by btwns7** [axisj/axisj#808]
소스는 아래와 같습니다.
AXGrid.js를 조금 수정해서 페이지네이게이션은 나오게 했는데, 실제 페이징이 안 되는 것 같습니다.

css, js 불러오는 부분.
`
    <\link rel="stylesheet" type="text/css" href="${pageContext.request.contextPath}/resources/axisj/ui/sh_aml/AXJ.css" />
    <\link rel="stylesheet" type="text/css" href="${pageContext.request.contextPath}/resources/axisj/ui/sh_aml/AXGrid.css" />
    <\link rel="stylesheet" type="text/css" href="${pageContext.request.contextPath}/resources/axisj/ui/sh_aml/AXButton.css" />
    <\link rel="stylesheet" type="text/css" href="${pageContext.request.contextPath}/resources/axisj/ui/sh_aml/AXInput.css" />
    <\link rel="stylesheet" type="text/css" href="${pageContext.request.contextPath}/resources/axisj/ui/sh_aml/AXSelect.css" />
    

    <script type="text/javascript" src="${pageContext.request.contextPath}/resources/axisj/dist/AXJ.min.js"></script>
    <script type="text/javascript" src="${pageContext.request.contextPath}/resources/axisj/lib/AXGrid.js"></script>
    <script type="text/javascript" src="${pageContext.request.contextPath}/resources/axisj/lib/AXInput.js"></script>
    <script type="text/javascript" src="${pageContext.request.contextPath}/resources/axisj/lib/AXSelect.js"></script>`

자바스크립트
`    var gridObj = new AXGrid();
    $(document).ready(function() {  
        $("#btn_search").on("click", function() {
            gridObj.reloadList();
        });
        
        fnObj.initGrid();   // 그리드 초기화
    });
    
    var fnObj = {
            initGrid: function() {

                gridObj.setConfig({
                    targetID : "prgrmAccessHistTable",
                    theme : "AXGrid",
                    fitToWidth:true,
                    colGroup : [
                        {key:"no",                  label:"no",         width:"40",  align:"center", formatter:"checkbox" },
                        {key:"EMP_NM",              label:"직원명",     width:"70"                                        },
                        {key:"PRGRM_NM",            label:"프로그램명", width:"100", align:"center"                       },
                        {key:"PRGRM_ARGS",          label:"변수",       width:"300", align:"left"                         },
                        {key:"PRGRM_ACCESS_DTTM",   label:"접근시간",   width:"100", align:"center", formatter:"text"     },
                        {key:"USER_IP",             label:"사용자 IP",  width:"100", align:"center", formatter:"text"     }
                    ],
                    page: {
                        paging: true,
                        pageNo: 1,
                        pageSize: 20
                    }
                });

                gridObj.setList({
                    ajaxUrl : ctx + "/sys/sys5200/getPrgrmAccessHist.do",
                    ajaxPars: $('#search_form').formToJson(),
                    onLoad  : function() {

                    }
                });
            },
            getCheckedList: function(){
                var checkedList = gridObj.getCheckedList(0);// colSeq
                alert(Object.toJSON(checkedList));
            },
            emptyList: function(){
                gridObj.setList({
                    ajaxUrl: ctx + "/sys/sys5200/getPrgrmAccessHist.do",
                    ajaxPars:"empty=true",
                    onLoad:function(){
                        // 로드 후...
                    }
                });

                //gridObj.setDataSet({price:0, amount:0});
            },
            reloadList: function() {
                gridObj.reloadList();
            }
        };
`

html body
`
<body>

    
    <form id="search_form" role="form" data-toggle="validator">
        <div class="searchWrap">
            <table class="w100">
                <tr>
                    <td>
                        <table>
                            <tr>
                                <th>직원명</th>
                                <td><input type="text" class="searchText" name="EMP_NM" /></td>
                                <th>프로그램명</th>
                                <td><input type="text" class="searchText" name="PRGRM_NM" /></td>
                                <th>날짜</th>
                                <td><input type="text" name="START_PRGRM_ACCESS_DTTM" id="dateObjSt" value=""
                                    class="date" /> <a href="#none" name="btn_cal"><img
                                        src="${pageContext.request.contextPath}/resources/images/common/btn_calendar.gif"
                                        alt="달력" /></a> ~ <input type="text" name="END_PRGRM_ACCESS_DTTM" id="dateObjEd"
                                    value="" class="date" /> <a href="#none" name="btn_cal"><img
                                        src="${pageContext.request.contextPath}/resources/images/common/btn_calendar.gif"
                                        alt="달력" /></a></td>
                                <th>IP</th>
                                <td><input type="text" class="searchText" name="USER_IP" /></td>
                            </tr>
                        </table>
                    </td>
                    <td class="middleRight"><a id="btn_search" href="#" class="btn search" >조회</a></td>
                </tr>
            </table>
        </div>
    </form>
    
    <br />
    <div id="prgrmAccessHistTable"></div>
</body>`

![image](https://cloud.githubusercontent.com/assets/6446916/12817474/84a8baca-cb95-11e5-8c98-3090bceed8b6.png)

**Comment by thomasJang** [comment]
setList로는 paging을 할 수 없습니다.

http://dev.axisj.com/samples/AXGrid/setData.html 여기 샘플을 체크 하세요.

**Comment by btwns7** [comment]
@thomasJang setData로 했더니, navigation이 표시는 되는데, 페이지가 실제로 이동 되지 않습니다.

setList는 ['ajax & paging'](http://dev.axisj.com/samples/AXGrid/ajax.html) 샘플을 참고해서 작성한 것인데, ajax를 사용하면서 페이징은 안되는 것인가요?

**Comment by lhslive** [comment]
서버 응답을 어떤 형태로 하고 계시나요?
pageNo , pageSize 를 정상적으로 넘겨 받아 처리하시면 특별히 문제 될건 없어 보입니다.

**Comment by btwns7** [comment]
@lhslive 서버에서 따로 처리하는 것은 없고 조회결과만 리스트로 받아오고 있습니다.
말씀하신 pageNo, pageSize를 정상적으로 처리하려면 어떻게 해야 하는지 알 수 있을까요?

**Comment by lhslive** [comment]
pageSize = 한페이지내에 보여줄 아이템 수
pageNo = 현 페이지.

이는 그리드에서 페이징 처리시 ajaxPars에 포함되어 날아가는 요소 입니다.
fiddler나 각 브라우저 개발자도구의 네트워크 항목을 보시면 도움이 되실 듯 합니다.
저같은 경우 두 요소를 가지고 쿼리할때 페이징 처리를 해줍니다.
기타 서버 프로그램에서 따로 페이징을 하신다면 이를 가지고 하시면 되겠죠.
mssql 같은 경우엔 
Row_Number() OVER (ORDER BY a.seq desc) AS rownum 를 가지고 
where 조건에 
rownum BETWEEN #start# AND #length# 를 줍니다.

start는 pageNo 
length 는 pageNo  * pageSize  를 주면 되겠지요..
pageNo 가 1이 아닐때는 (pageNo - 1) * pageSize + 1
식으로 커스텀 페이징을 구현해 주시면 됩니다. 
그리고 응답은 추가적으로 
http://dev.axisj.com/samples/AXGrid/ajax.html 
이 링크를 개발자도구로 보시면 네트워크 항목에 응답형태가 나오니 참고 하시면 더 빠르시리라 생각됩니다.
```

{
	list:[
            {no:1, no2:1, title:"AXGrid data line 1/1", img:"images/1.jpg", writer:"Thomas", regDate:"2013-01-18", desc:"myGrid.setList", price:123000, amount:10}
            ,{no:2, no2:2, title:"AXGrid data line 1/2", img:"images/2.jpg", writer:"Thomas", regDate:"2013-01-18", desc:"myGrid.setList", price:123000, amount:10}
            ,{no:3, no2:3, title:"AXGrid data line 1/3", img:"images/3.jpg", writer:"Thomas", regDate:"2013-01-18", desc:"myGrid.setList", price:123000, amount:10}
            ,{no:4, no2:4, title:"AXGrid data line 1/4", img:"images/4.jpg", writer:"Thomas", regDate:"2013-01-18", desc:"myGrid.setList", price:123000, amount:10}
 
    	],
	page:{
		pageNo:1,
		pageCount:200,
		listCount:100
	},
	msg:""
}
```
